spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/KafkaConfig.java

```diff
@@ -128,31 +183,37 @@ public Schema getSchema() {
     }
   }
 
-  // gets the message schema from the schema field. If the time field and key field are in the configured
+  // gets the message schema from the schema field. If the time, key, partition, or offset fields are in the configured
   // schema, they will be removed.
   public Schema getMessageSchema() {
     Schema schema = getSchema();
     List<Schema.Field> messageFields = new ArrayList<>();
     boolean timeFieldExists = false;
     boolean keyFieldExists = false;
+    boolean partitionFieldExists = false;
+    boolean offsetFieldExists;
+
     for (Schema.Field field : schema.getFields()) {
       String fieldName = field.getName();
       Schema fieldSchema = field.getSchema();
+      Schema.Type fieldType = fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
       // if the field is not the time field and not the key field, it is a message field.
       if (fieldName.equals(timeField)) {
-        Schema.Type timeType = fieldSchema.isNullable() ?
-          fieldSchema.getNonNullable().getType() : fieldSchema.getType();
-        if (timeType != Schema.Type.LONG) {
+        if (fieldType != Schema.Type.LONG) {
           throw new IllegalArgumentException("The time field must be of type long or nullable long.");
         }
         timeFieldExists = true;
       } else if (fieldName.equals(keyField)) {
-        Schema.Type timeType = fieldSchema.isNullable() ?
-          fieldSchema.getNonNullable().getType() : fieldSchema.getType();
-        if (timeType != Schema.Type.BYTES) {
+        if (fieldType != Schema.Type.BYTES) {
           throw new IllegalArgumentException("The key field must be of type bytes or nullable bytes.");
         }
         keyFieldExists = true;
+      } else if (fieldName.equals(partitionField)) {
+        if (fieldType != Schema.Type.INT) {
+          throw new IllegalArgumentException("The partition field must be of type int.");
+        }
+        partitionFieldExists = true;
+      } else if (fieldName.equals(offsetField)) {
```

Comment:
Do we need to do any check for offsetField? If not can we remove this "else if (fieldName.equals(offsetField))"